controllers/pod.go

```diff
@@ -345,9 +350,14 @@ func (r *SingleClusterReconciler) ensurePodsRunningAndReady(podsToCheck []*corev
 	return reconcileRequeueAfter(10)
 }
 
-func getFailedAndActivePods(pods []*corev1.Pod) (failedPods, activePods []*corev1.Pod) {
+func getFailedAndActivePods(pods []*corev1.Pod, ignorablePodNames sets.Set[string],
+) (failedPods, activePods []*corev1.Pod) {
 	for idx := range pods {
 		pod := pods[idx]
+		if ignorablePodNames.Has(pod.Name) {
```

Comment:
we calls `getFailedAndActivePods` at 4 places, at 2 places we make sure that there are no ignorable pods present in pods list passed to this function, and in other two places from `reconcileRacks()`, there are chances of having ignorable pods in pods list. Can we write it better and remove check `if ignorablePodNames.Has(pod.Name) {` from `getFailedAndActivePods` if possible.